client/signup/steps/survey/survey-step-v2.jsx

```diff
@@ -107,7 +107,7 @@ export default React.createClass( {
 
 	handleOtherWriteIn( e ) {
 		this.setState( {
-			otherWriteIn: e.target.value
+			otherWriteIn: e.target.value.replace( /^(\s|\.|\,|\:|\!|\?)+|(\s|\.|\,|\:|\!|\?)+$/g, '' ),
```

Comment:
Also, do we have polyfills in place to allow the `u` ES6 RegExp flag? Would be useful for i18n.